src/randgraphs.jl

```diff
@@ -107,9 +107,13 @@ function _suitable(edges::Set{Edge}, potential_edges::Dict{Int, Int})
     return false
 end
 
-function _try_creation(n::Int, k::Int)
+function _try_creation(n::Int,@compat k::Union{Int,Array{Int}})
```

Comment:
This will be faster if we specialize on the type. Can we have two `_try_creation` methods instead?

```
function _try_creation(n::Int, k::Int)
...
function _try_creation(n::Int, k::Vector{Int})
...
```
?